kart/sqlalchemy/adapter/gpkg.py

```diff
@@ -529,6 +533,33 @@ def gpkg_to_xml_metadata(cls, gpkg_meta_items):
             AND MR.row_id_value IS NULL;
         """
 
+    @classmethod
+    def find_sole_useful_xml(cls, xml_list):
+        """
+        Given a list containing more than one piece of metadata XML, returns the only piece XML that is actually useful,
+        if one can be decided on using a heuristic.
+        """
+        # Currently only the following heuristic is implemented:
+        # If there are two pieces of XML and the second piece of XML is the same as the first XML, but wrapped inside a
+        # `<GDALMultiDomainMetadata>` structure, then we just return the first piece of XML (or vice versa).
+        # In all other cases we have no opinion on which piece(s) of XML are useful or useless.
+        if len(xml_list) != 2:
+            return None
+        from html import unescape
+
+        for first, second in ((xml_list[0], xml_list[1]), (xml_list[1], xml_list[0])):
+            wrapper = unescape(second).split(first, maxsplit=1)
```

Comment:
Wouldn't it be just as simple to use an XML parser? H̸̡̪̯ͨ͊̽̅̾̎Ȩ̬̩̾͛ͪ̈́̀́͘ ̶̧̨̱̹̭̯ͧ̾ͬC̷̙̲̝͖ͭ̏ͥͮ͟Oͮ͏̮̪̝͍M̲̖͊̒ͪͩͬ̚̚͜Ȇ̴̟̟͙̞ͩ͌͝S̨̥̫͎̭ͯ̿̔̀ͅ ([xml.etree in the stdlib](https://docs.python.org/3/library/xml.etree.elementtree.html), don't bother with the other stdlib ones)

Reply:
Let's say I'm doing test driven development and I only write code if it helps the test to pass. We don't care enough about the content of this XML for me to write any more tests, and so the regex approach is probably the least amount of code I can write and should be faster than parsing a tree for this kind of check. I can always swap it out later if we start digging deeper into XML contents